secboot/secboot_tpm_test.go

```diff
@@ -148,36 +160,47 @@ func (s *secbootSuite) TestMeasureSnapSystemEpochWhenPossible(c *C) {
 }
 
 func (s *secbootSuite) TestMeasureSnapModelWhenPossible(c *C) {
-	for _, tc := range []struct {
-		tpmErr   error
-		modelErr error
-		callNum  int
-		err      string
+	for i, tc := range []struct {
+		tpmErr     error
+		tpmEnabled bool
+		modelErr   error
+		callNum    int
+		err        string
 	}{
 		{
 			// normal connection to the TPM device
-			tpmErr: nil, modelErr: nil, callNum: 1, err: "",
+			tpmErr: nil, tpmEnabled: true, modelErr: nil, callNum: 1, err: "",
 		},
 		{
 			// normal connection to the TPM device with model error
-			tpmErr: nil, modelErr: errors.New("model error"), callNum: 0,
+			tpmErr: nil, tpmEnabled: true, modelErr: errors.New("model error"), callNum: 0,
 			err: "cannot measure snap model: model error",
 		},
 		{
 			// TPM device exists but returns error
 			tpmErr: errors.New("tpm error"), callNum: 0,
 			err: "cannot measure snap model: cannot open TPM connection: tpm error",
 		},
+		{
+			// TPM device exists but is disabled
+			tpmErr: nil, tpmEnabled: false,
+		},
 		{
 			// TPM device does not exist
-			tpmErr: &os.PathError{Op: "open", Path: "path", Err: errors.New("enoent")}, callNum: 0, err: "",
+			tpmErr: sb.ErrNoTPM2Device,
 		},
 	} {
+		c.Logf("%d: tpmErr:%v tpmEnabled:%v", i, tc.tpmErr, tc.tpmEnabled)
 		mockModel := &asserts.Model{}
 
 		mockTpm, restore := mockSbTPMConnection(c, tc.tpmErr)
 		defer restore()
 
+		restore = secboot.MockIsTPMEnabled(func(tpm *sb.TPMConnection) bool {
+			return tc.tpmEnabled
```

Comment:
Wondering whether it makes sense to verify that this was never called when ErrNoTPM2Device was returned.

Reply:
This case would be a programming error where we call tpm.IsEnabled() for tpm == nil and we would see many tests crashing.